mongodbatlas/resource_mongodbatlas_cluster.go

```diff
@@ -346,13 +365,15 @@ func resourceMongoDBAtlasClusterCreate(d *schema.ResourceData, meta interface{})
 	providerSettings := expandProviderSetting(d)
 
 	replicationSpecs, err := expandReplicationSpecs(d)
-
 	if err != nil {
 		return fmt.Errorf(errorCreate, err)
 	}
 
-	autoScaling := matlas.AutoScaling{
-		DiskGBEnabled: pointy.Bool(d.Get("auto_scaling_disk_gb_enabled").(bool)),
+	// set the default disk size gb for the provider and tier
+	diskSizeGB := matlas.DefaultDiskSizeGB[providerName][d.Get("provider_instance_size_name").(string)]
+	// But if it's specify the disk size will set up
```

Comment:
Just be aware that if someone does specify it and it's lower than the default Atlas automatically sets it to the default.